packages/cdk-assets/lib/private/docker.ts

```diff
@@ -70,9 +86,49 @@ export class Docker {
     await this.execute(['push', tag]);
   }
 
+  /**
+   * If a CDK Docker Credentials file exists, creates a new Docker config directory.
+   * Sets up `docker-credential-cdk-assets` to be the credential helper for each domain in the CDK config.
+   * All future commands (e.g., `build`, `push`) will use this config.
+   *
+   * See https://docs.docker.com/engine/reference/commandline/login/#credential-helpers for more details on cred helpers.
+   *
+   * @returns true if CDK config was found and configured, false otherwise
+   */
+  public configureCdkCredentials(): boolean {
+    const config = cdkCredentialsConfig();
+    if (!config) { return false; }
+
+    const configDir = this.createCleanConfig();
+
+    const domains = Object.keys(config.domainCredentials);
+    const credHelpers = domains.reduce(function(map: Record<string, string>, domain) {
+      map[domain] = 'cdk-assets'; // Use docker-credential-cdk-assets for this domain
+      return map;
+    }, {});
+    fs.writeFileSync(path.join(configDir, 'config.json'), JSON.stringify({ credHelpers }), { encoding: 'utf-8' });
+
+    return true;
+  }
+
+  /**
+   * Creates a new empty Docker config directory.
+   * All future commands (e.g., `build`, `push`) will use this config.
+   *
+   * This is useful after calling `configureCdkCredentials` to reset to default credentials.
+   *
+   * @returns the path to the directory
+   */
+  public createCleanConfig(): string {
+    this.configDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdkDockerConfig'));
+    return this.configDir;
```

Comment:
Nit: this method is violating [command-query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation), and not even for a good reason (such as performing an atomic operation or something). Couldn't the caller just read `this.configDir` after calling this?

Reply:
Yeah, you're right. I was being lazy. :(. On line 102 above, I was calling this method, and then needing to check that `this.configDir` was defined before using it on line 109, which felt funny. So I cheated and made this method return the path. I'll revert that.